### ISSUE TITLE
Ignore broken chainlinks

### ISSUE THREAD

**Issue by maxtaco** [keybase/libkeybase-js#5]
Include a hardcoded list of broken chainlinks, and check they fit into the sigchain properly, but otherwise discard them and don't consider their contents.

cc: @akalin-keybase whose sigchain is broken

Two chainlinks to add to the list:
```json
[
    "2a0da9730f049133ce728ba30de8c91b6658b7a375e82c4b3528d7ddb1a21f7a0f",
    "0f175ef0d3b57a9991db5deb30f2432a85bc05922bbe727016f3fb660863a1890f"
]
```

**Comment by oconnor663** [comment]
Will this list end up getting shared with the Go client?

**Comment by maxtaco** [comment]
yes

**Comment by akalin-keybase** [comment]
There are actually a total of four links to remove:

```
// A template for some of the reasons in badChainLinks below.
const badLinkTemplate = "Link %d of akalin's sigchain, which was accidentally added by an old client in development on 23 Mar 2015 20:02 GMT."

// A map from SigIDs of bad chain links that should be ignored to the
// reasons why they're ignored.
var badChainLinks = map[keybase1.SigID]string{
       // Links 22-25 of akalin's sigchain, which was accidentally
       // added by an old client in development on 3/23/2015, 9:02am.
       "2a0da9730f049133ce728ba30de8c91b6658b7a375e82c4b3528d7ddb1a21f7a0f": fmt.Sprintf(badLinkTemplate, 22),
       "eb5c7e7d3cf8370bed8ab55c0d8833ce9d74fd2c614cf2cd2d4c30feca4518fa0f": fmt.Sprintf(badLinkTemplate, 23),
       "0f175ef0d3b57a9991db5deb30f2432a85bc05922bbe727016f3fb660863a1890f": fmt.Sprintf(badLinkTemplate, 24),
       "48267f0e3484b2f97859829503e20c2f598529b42c1d840a8fc1eceda71458400f": fmt.Sprintf(badLinkTemplate, 25),
}
```

**Comment by akalin-keybase** [comment]
Andy gets:

```
$ keybase encrypt akalin
error: read off end of buffer
```

which may be related to this, perhaps.

**Comment by maxtaco** [comment]
Can you run with `-d` ?

**Comment by aalness** [comment]
[fail_log.txt.gz | uploaded via ZenHub](https://files.zenhub.io/5600b8fc083043f44e2af480)

**Comment by maxtaco** [comment]
Ah cool.  Of course, haven't ported these bad links to node yet.  That's almost certainly the issue.  Thanks Andy.

**Comment by maxtaco** [comment]
Yes, @akalin-keybase, exactly related.

**Comment by akalin-keybase** [comment]
I might be have some bandwidth to do this, if @oconnor663 is busy with other stuff. :)

**Comment by maxtaco** [comment]
Sure sounds good.  We should also add your sigchain as a test vector to `keybase-test-vectors`

**Comment by oconnor663** [comment]
shipped in node client version 0.8.22
